packages/office-ui-fabric-react/src/components/Pivot/Pivot.base.tsx

```diff
@@ -192,38 +178,64 @@ export class PivotBase extends BaseComponent<IPivotProps, IPivotState> {
     );
   };
 
+  private _selectedKey(): string | undefined {
+    if (this._isKeyValid(this.props.selectedKey)) {
+      return this.props.selectedKey!;
+    }
+    if (this._isKeyValid(this.state.selectedKey)) {
+      return this.state.selectedKey;
+    }
+    if (this._links.length) {
+      return this._links[0].itemKey!;
+    }
+  }
+
+  private _isItemSelected(itemKey: string | undefined): boolean {
+    return this._selectedKey() === itemKey;
+  }
+
+  private _selectedTabId(): string | undefined {
+    const selectedKey = this._selectedKey();
+    if (selectedKey) {
+      return this._keyToTabIds[selectedKey];
+    }
+    return undefined;
+  }
+
   /**
    * Gets the set of PivotLinks as arrary of IPivotItemProps
    * The set of Links is determined by child components of type PivotItem
    */
-  private _getPivotLinks(props: IPivotProps): IPivotItemProps[] {
+  private _updatePivotLinks(): void {
```

Comment:
Curious of your reasoning for the change from returning links to updating the private member within the method?